File: Java-version/Guarana.java
import java.awt.*;

public class Guarana extends Plant{
    public Guarana(int x, int y, World world)
    {super(x,y,world);
        sign = 'U';
        name = "Guarana";
        color = Color.GRAY;
        str = 0;
        speciesId = 7;
        spreadChance = 40;
        spreadCount = 1;
    }
    public Guarana(int str, int ini, int xPos, int yPos, int age, int oId, boolean moved, World world)
    {super(str, ini, xPos, yPos, age, oId, moved, world);
        sign = 'U';
        name = "Guarana";
        speciesId = 7;
        spreadChance = 10;
        spreadCount = 1;
        color = Color.GRAY;
    }
    @Override
    public Plant MakeNewP(int x, int y) {
        Plant newPlant = new Guarana(x, y, world);
        return newPlant;
    }
    @Override
    public boolean Defended(Organism attackingCreature) {
        int newStr = attackingCreature.GetS() + 3;
        attackingCreature.SetS(newStr);
        System.out.print(", which is granting strength upon being eaten,");
        if (str > attackingCreature.GetS())
            return true;
        return false;
    }
}

File: Python-version/guarana.py
from plant import Plant

class Guarana(Plant):
    def __init__(self, x, y, world, oid=None, s=None, i=None, a=None, m=None):
        super().__init__(x, y, world, oid, str, i, a, m)
        self.color = "gray"
        self.sign = 'U'
        self.spread_count = 1
        self.spread_chance = 25
        self.str=0
        self.speciesId=7
        self.ini=0
        self.name="Guarana"

    def MakeNewP(self,x,y):
        return Guarana(x,y,self.world)

    def Defended(self,attackingcreature):
        new_strength=attackingcreature.GetS()+3
        attackingcreature.SetS(new_strength)
        print(f", which is granting strength upon being eaten")
        if self.str>attackingcreature.GetS():
            return True
        return False